### DISCUSSION TITLE
Weird issue with Modal + Heap event definitions

### DISCUSSION THREAD

**Discussion by rosemaryluo** [twilio-labs/paste#2947] [General] [answered]
Hi all, I'm having an issue with interacting with the Heap event definition controls when defining events on a modal. I'm wondering if this is because 

Modal guidance: _"Users cannot interact with content outside an active modal window until the user explicitly closes it. When the modal opens, focus moves to an element contained in it (by default, the first element that can be focused) in the modal, and the focus should stay in and cycle through the modal's content. Focus shouldn't return to the underlying page until the user closes the modal."_

And wondering if there are any easy ways to dodge this while defining events.

<img width="1249" alt="image" src="https://user-images.githubusercontent.com/87049408/212406247-8f0bb00e-203d-4bea-951f-a9c53f1aedb0.png">

**Comment by nkrantz** [comment]
Hi @rosemaryluo! Chatting with the team about this. We'll get back to you after the long weekend. Cheers!

**Comment by nkrantz** [comment]
Hi again! Just want to be sure - are you having trouble tracking events in the Modal, or events on the UI behind the Modal overlay?

**Reply by rosemaryluo** [reply]
In the modal. I can click links in the heap panel, but I can't scroll the Edit Definition section (left and right) and cannot input text--like if I try to edit the Limit to URL field or other fields on the next page of the definition wizard.

**Comment by loreina** [comment, marked as answer]
Hi @rosemaryluo, unfortunately this falls outside of the realm of what we typically provide support on.

We'd suggest asking in the #help-design-system or #eng-guild-frontend channels in Slack to see if anyone else has run into similar problems with modals and event tracking. 

Hope you find a solution soon!

**Reply by rosemaryluo** [reply]
I'll look into a workaround, I was just wondering whether this had to do with how the Paste modal works. Thank you!
